### PR TITLE
Ensure consistent code conventions

### PR DESCRIPTION
**Pull request by magro**
This addresses #132 - adds both editorconfig and mega-linter setup, for details see the different commit messages.

Changes might be needed, all written on the phone, I'm curious to see how it works 😁

### CODE REVIEW

**Comment by magro** [comment]
Hmm, the broken conventions were not detected, doesn't seem to work and needs more investigation.

Note: MegaLinter edited the original comment (https://github.com/tomorrow-one/transactional-outbox/pull/133#issuecomment-1612102715) after the run for the test commit breaking conventions, instead of adding a new comment. The order of comments might be confusing...

**Comment by magro** [comment]
Ok, it works, the only thing is that https://github.com/editorconfig-checker/editorconfig-checker#what states that it only supports certain editorconfig features:
* `end_of_line`
* `insert_final_newline`
* `trim_trailing_whitespace`
* `indent_style`
* `indent_size`
* `max_line_length`

while in my previous test yesterday I broke `spaces_around_operators` (in commit 73f5a5aa).

Now breaking the `indent_style` and `trim_trailing_whitespace` properties shows that at least these properties are checked.

Not sure, should we integrate this, @hnrkdmsk @danielrehmann ?

File: .mega-linter.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+ENABLE:
+  - EDITORCONFIG
```
Comment (hnrkdmsk) [inline .mega-linter.yml:2]: I've a question regarding this: Based on the [documentation](https://megalinter.io/latest/descriptors/editorconfig_editorconfig_checker/), you have to set `EDITORCONFIG_EDITORCONFIG_CHECKER ` here, or I'm in the documentation? 🤔

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,21 @@
+root = true
```
Comment (hnrkdmsk) [inline .editorconfig:1]: We're trying to use the mega linter together with the editorconfig-checker, right? If so, then I think this configuration is not what we need based on the [documentation](https://editorconfig-checker.github.io) 🤔

**Review by hnrkdmsk — COMMENTED** [review]
Maybe we can try out what I've seen in the documentation if you didn't already 🤓

File: .mega-linter.yml (hunk 1 above)
Comment (magro) [reply .mega-linter.yml:2]: Well, we've seen that it's doing its job, haven't we? Not sure what this should change.

File: .editorconfig (hunk 2 above)
Comment (magro) [reply .editorconfig:1]: Can you point out to what exactly you're referring to and quote that please?

Or just feel free to experiment and change in your own if you see potential improvements, maybe that's more efficient.

Comment (hnrkdmsk) [reply .editorconfig:1]: > Configuration
> 
> There are a bunch of command line flags you can use, but there is also a configuration file which you can use in  your current working directory called .ecrc. If you want to use a different config you can pass the -config <path>- flag to the binary. You can generate a configuration with the init -flag. A sample configuration file can look like this:
> 
> {
>   "Verbose": false,
>   "Debug": false,
>   "IgnoreDefaults": false,
>   "SpacesAftertabs": false,
>   "NoColor": false,
>   "Exclude": [],
>   "AllowedContentTypes": [],
>   "PassedFiles": [],
>   "Disable": {
>     "EndOfLine": false,
>     "Indentation": false,
>     "InsertFinalNewline": false,
>     "TrimTrailingWhitespace": false
>   }
> }

Sorry, this is what I'm talking about 🤗 I can of course experiment with this 👍🏼

Comment (magro) [reply .editorconfig:1]: As I understand it this is an alternative to the configuration via `.editorconfig`. As said, the `.editorconfig` is taken into account, not sure if you've seen it. I have the feeling that we're on quite different pages somehow...

Comment (hnrkdmsk) [reply .editorconfig:1]: Ahh 💡, after a deeper look into the configuration and the logs of the GitHub Actions, I did understand that it is doing what it has to do 👍🏼 

For me this is a good start, let's merge it 🤗 Thanks for implementing this @magro!

**Review by hnrkdmsk — APPROVED** [review]

**Comment by hnrkdmsk** [comment]
> Ok, it works, the only thing is that https://github.com/editorconfig-checker/editorconfig-checker#what states that it only supports certain editorconfig features:
> 
> * `end_of_line`
> * `insert_final_newline`
> * `trim_trailing_whitespace`
> * `indent_style`
> * `indent_size`
> * `max_line_length`
> 
> while in my previous test yesterday I broke `spaces_around_operators` (in commit [73f5a5a](https://github.com/tomorrow-one/transactional-outbox/commit/73f5a5aaf61258655031907e66485ebb6b37ed12)).
> 
> Now breaking the `indent_style` and `trim_trailing_whitespace` properties shows that at least these properties are checked.
> 
> Not sure, should we integrate this, @hnrkdmsk @danielrehmann ?

I think we should use this. It is working, and we can still improve it if we have the need. It's better than nothing 😊

**Pull request merged**